### PR TITLE
Fix AGR generation errors by making it work more like RO

### PR DESCRIPTION
**Pull request by fenhl**
**Edit:** [See this comment for the new fix.](https://github.com/TestRunnerSRL/OoT-Randomizer/pull/1441#issuecomment-973643433)

Updates the fill code to ensure that once it has started to do access checks, it will keep doing them for the rest of the fill. This seems to fix the generation failure with this plando:

```json
{
    "settings": {
        "reachable_locations": "goals",
        "bridge_medallions": 4,
        "shuffle_ganon_bosskey": "vanilla"
    }
}
```

It should also be a minor seed generation speed-up in AGR and RO since it won't do extra `can_beat_game` checks anymore to determine whether to do access checks if it's already doing access checks.

### CODE REVIEW

**Comment by fenhl** [comment]
As discussed in #dev-bug-reports, this is just a quick fix to make these seeds generate again, with the possible (untested) side effect of AGR behaving too similar to ALR. A proper fix might be to remove the `perform_access_check` and `extra_location_checks` distinctions entirely (or modify them to treat AGR like RO) and instead replace all `can_beat_game` checks with an equivalent `can_beat_goals` in AGR.

**Review by Zannick — APPROVED** [review]

**Comment by fenhl** [comment]
I updated this with a different fix that makes AGR work essentially like RO, except the condition for the accessibility check when an item is being placed is that there are unreachable goals without it rather than that the game is unbeatable without it.

The previous fix was based on an inaccurate comment, which I've now removed. I also renamed some functions related to beatable goals to better reflect that they return `dict`, not `bool`, and updated the setting description for Bridge and Gbk no longer being goals in TH, a change made in #1361.

File: SettingsList.py [hunk 1]
```diff
@@ -2236,7 +2236,7 @@ def __init__(self, name, gui_text, min, max, default, step=1,
             to beat the game, but otherwise behaves like 'Required Only'.
             Goal items are the items required for the rainbow bridge and/or Ganon's Boss Key, so for example if the bridge is
             set to 1 Medallion and Ganon's Boss Key to 1 Gold Skulltula Token, all 6 Medallions and all 100 Tokens will
-            be obtainable. In Triforce Hunt, this will also guarantee that all Triforce Pieces can be obtained.
+            be obtainable. In Triforce Hunt, this will instead guarantee that all Triforce Pieces can be obtained.
```
Comment (Zannick) [inline SettingsList.py:2239]: Why this change?

File: Search.py [hunk 2]
```diff
@@ -220,10 +220,10 @@ def progression_locations(self):
     #
     # Win condition can be a string that gets mapped to a function(state_list) here
     # or just a function(state_list)
```
Comment (Zannick) [inline Search.py:222]: This comment seems out of date as well.
```suggestion
    # predicate must be a function (state) -> bool, that will be applied to all states
```

File: SettingsList.py (hunk 1 above)
Comment (Zannick) [reply SettingsList.py:2239]: > Bridge and Gbk no longer being goals in TH, a change made in #1361.
Got it.

**Review by Zannick — APPROVED** [review]
